### ISSUE TITLE
How to resize items and get correct push behavior

### ISSUE THREAD

**Issue by MgSam** [tiberiuzuld/angular-gridster2#446]
We need to programmatically resize a gridster item and have the gridster items below it appropriately pushed down (they should not resize themselves in response to the item above resizing). Currently, they are being pushed to the right instead, which is not the behavior we want. 

I've tried looking around this site but the documentation on `GridsterPushResize` is very sparse. I'm not actually sure what the difference between it and `GridsterPush` even is.

What is currently happening:
![image](https://user-images.githubusercontent.com/1965570/49596065-783c1d80-f947-11e8-841a-79272813edb5.png)

What I want to happen:
![image](https://user-images.githubusercontent.com/1965570/49596088-82f6b280-f947-11e8-9874-74a04679420d.png)

Here is my code:
``` TypeScript
let itemToPush = this.gridster.grid.find(i => i.item.config === t.config);
const initialRowCount = itemToPush.$item.rows;

if (initialRowCount === newRowCount) { return; }
itemToPush.$item.rows = newRowCount;

const push = new GridsterPushResize(itemToPush);
if (push.pushItems(push.fromNorth)) {
    push.checkPushBack();
    push.setPushedItems();
    itemToPush.setSize();
    itemToPush.checkItemChanges(itemToPush.$item, itemToPush.item);
} else {
    itemToPush.$item.rows = initialRowCount;
    push.restoreItems();
}
push.destroy();
```

Thanks for any help.

**Comment by tiberiuzuld** [comment]
Hi @MgSam ,
Your code should work, don't know why is not working.
Possibly a bug. 
Currently the PUSH api is not the best, I plan to change it when will have the time to make the improvements for it. https://github.com/tiberiuzuld/angular-gridster2/issues/331

**Comment by tiberiuzuld** [comment]
The issue should be fixed now in v10.1.2
